DIPs/DIP1NNN-DK.md

```diff
@@ -0,0 +1,696 @@
+# @system variables
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | (number/id -- assigned by DIP Manager)                          |
+| Review Count:   | 0                                                               |
+| Author:         | Dennis Korpel dkorpel@gmail.com                                 |
+| Implementation: |                                                                 |
+| Status:         |                                                                 |
+
+## Abstract
+When the `@system` attribute is attached to variables or fields, it means they cannot be written to in `@safe` functions.
+If the variable has an unsafe type (type that has at least one pointer or `@system` field), it also cannot be read in `@safe` code.
+This allows more safe encapsulation of low-level unsafe code using `@trusted`, since currently `@trusted` code cannot assume anything about the integrity of data.
+It also fixes some existing issues with `@safe` code.
+
+## Contents
+* [Background](#background)
+* [Rationale](#rationale)
+* [Prior work](#prior-work)
+* [Description](#description)
+* [Alternatives](#alternatives)
+* [Breaking Changes and Deprecations](#breaking-changes-and-deprecations)
+* [Reference](#reference)
+* [Copyright & License](#copyright--license)
+* [Reviews](#reviews)
+
+## Background
+
+A distinction can be made between syntactic types and semantic types.
+Syntactic types are the things the compiler understands: you can't assign a string to an int, you can't pass an integer array to a function that takes a float parameter etc.
+While this helps catching bugs that assembly programmers encounter, it has its limitations with regards to data abstraction and unsafe features.
+Often it is useful to make types that have additional constraints on the primitive types they consist of, or types that from the outside are safe and type-sound, but internally use unsafe operations that circumvent the type system.
+This can only work if the language in which such types are defined has the needed encapsulation capabilities.
```

Comment:
I don't understand the background section or what it's trying to accomplish.

Reply:
The mentioned article gives some high level rationale for what this DIP is trying to accomplish (i.e. making the type system catch more errors than simple type mismatches) and the section is a small summary of it.